File: 프로그래머스/lv0/120852. 소인수분해/소인수분해.py
def isPrime(x):
    for i in range(2,x):
        if x%i==0:
            return False
    return True

def solution(n):
    answer = []
    for i in range(2,n+1):
        if n%i==0 and isPrime(i):
            answer.append(i)
    return answer

File: 프로그래머스/lv0/120852. 소인수분해/소인수분해.java
import java.util.*;

class Solution {
    public int[] solution(int n) {
        ArrayList<Integer> result = new ArrayList<>();        
        for(int i=2; i<=n; i++){
            if (n%i==0){
                result.add(i);
                n /= i;
            }
            while(n%i==0) {
                n /= i;
            }
        }
        
        int[] answer = new int[result.size()];
        for(int i=0; i<result.size(); i++){
            answer[i] = result.get(i);
        }
 
    


        return answer;
    }
}